src/collector.rs

```diff
@@ -61,79 +61,79 @@ struct RecorderHandle(mpsc::Sender<Datum>);
 
 struct Collector {
     metrics_data: BTreeMap<Timestamp, HashMap<(Key, Kind), StatisticSet>>,
-    config: Config,
+    config: &'static Config,
 }
 
 pub fn init(config: Config) {
     let _ = thread::spawn(|| {
-        tokio::runtime::current_thread::run(init_future(config).map_err(|e| {
-            log::warn!("{}", e);
-        }));
+        let mut runtime = tokio::runtime::Builder::new()
+            // single threaded
+            .basic_scheduler()
```

Comment:
You may need to `enable_time` `enable_io` https://docs.rs/tokio/0.2.11/tokio/runtime/struct.Builder.html#method.enable_time

Reply:
Just discovered that.. thanks for catching it!